modules/local/higher_level_force_config.nf

```diff
@@ -0,0 +1,83 @@
+nextflow.enable.dsl = 2
+
+process HIGHER_LEVEL_CONFIG {
+
+    label 'process_single'
+    tag { tile }
+
+    container "docker.io/davidfrantz/force:3.7.10"
+
+    input:
+    tuple val(tile), path("ard/${tile}/*"), path("ard/${tile}/*"), path("mask/${tile}/aoi.tif")
+    path 'ard/datacube-definition.prj'
+    path endmember
+
+    output:
+    tuple val (tile), path("trend_${tile}.prm"), path("ard/", includeInputs: true), path("mask/", includeInputs: true), path('ard/datacube-definition.prj', includeInputs: true), path(endmember, includeInputs: true), emit: higher_level_configs_and_data
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    """
+    # generate parameterfile from scratch
+    force-parameter -c ./trend_${tile}.prm TSA
+    PARAM=trend_"$tile".prm
+
+    # set parameters
+
+    #Replace pathes
+    sed -i "/^DIR_LOWER /c\\DIR_LOWER = ard/" \$PARAM
+    sed -i "/^DIR_HIGHER /c\\DIR_HIGHER = trend/" \$PARAM
+    sed -i "/^DIR_MASK /c\\DIR_MASK = mask/" \$PARAM
+    sed -i "/^BASE_MASK /c\\BASE_MASK = aoi.tif" \$PARAM
+    sed -i "/^FILE_ENDMEM /c\\FILE_ENDMEM = $endmember" \$PARAM
+
+    # threading
+    sed -i "/^NTHREAD_READ /c\\NTHREAD_READ = 1" \$PARAM              # might need some modification
+    sed -i "/^NTHREAD_COMPUTE /c\\NTHREAD_COMPUTE = $params.force_threads" \$PARAM  # might need some modification
+    sed -i "/^NTHREAD_WRITE /c\\NTHREAD_WRITE = 1" \$PARAM            # might need some modification
+
+
+    # replace Tile to process
+    TILE="$tile"
+    X=\${TILE:1:4}
+    Y=\${TILE:7:11}
+    sed -i "/^X_TILE_RANGE /c\\X_TILE_RANGE = \$X \$X" \$PARAM
+    sed -i "/^Y_TILE_RANGE /c\\Y_TILE_RANGE = \$Y \$Y" \$PARAM
+
+    # resolution
+    sed -i "/^RESOLUTION /c\\RESOLUTION = $params.resolution" \$PARAM
```

Comment:
As above, I wouldn't call parameters directly within a module, as they are look mandatory here,  I would put them as input cahnnels, and you call the parameter in the module invocation.

This gives you more flexibility to rename parameters and not having to hunt through every single module file (you just have to change in the main script or modules.conf)